beacon-chain/casper/incentives.go

```diff
@@ -18,7 +18,8 @@ func CalculateRewards(
 	voterIndices []uint32,
 	validators []*pb.ValidatorRecord,
 	totalParticipatedDeposit uint64,
-	timeSinceFinality uint64) []*pb.ValidatorRecord {
+	timeSinceFinality uint64) (newValidators []*pb.ValidatorRecord) {
```

Comment:
so this does not make it pure , you are still modifying the same underlying validator record. Assigning to a new variable does not change that

Reply:
I think this needs to be a deep copy?